ydb/library/yql/dq/comp_nodes/hash_join_utils/neumann_hash_table.h

```diff
@@ -434,14 +448,14 @@ class TNeumannHashTable {
 
         return iters;
     }
-    bool Empty(){
+    bool Empty() const{
         return Buffer_.empty();
     }
 
     void Apply(const ui8 *const row, const ui8 *const overflow,
                std::invocable<const ui8*> auto onMatch) const {
-        Y_ASSERT(Layout_ != nullptr);
-        Y_ASSERT(!Directories_.empty() && Tuples_ != nullptr);
+        MKQL_ENSURE(Layout_ != nullptr, "sanity check");
+        MKQL_ENSURE(!Directories_.empty() && Tuples_ != nullptr, "lookup to empty table?");
```

Comment:
If empty table lookup is not allowed there should be addition check, that in case of empty side we never build HT and never lookup